android/src/main/kotlin/com/hypertrack/sdk/flutter/common/Serialization.kt

```diff
@@ -0,0 +1,196 @@
+            KEY_TYPE to TYPE_IS_TRACKING,
+            KEY_VALUE to isTracking
+        )
+    }
+
+    fun serializeIsAvailable(isAvailable: Boolean): Map<String, Any?> {
+        return mapOf(
+            KEY_TYPE to TYPE_AVAILABILITY,
+            KEY_VALUE to isAvailable
+        )
+    }
+
+    fun serializeLocation(location: Location): Map<String, Any?> {
+        return mapOf(
+            KEY_TYPE to TYPE_LOCATION,
+            KEY_VALUE to mapOf(
+                KEY_LATITUDE to location.latitude,
+                KEY_LONGITUDE to location.longitude
+            )
+        )
+    }
+
+    fun serializeLocationError(locationError: LocationError): Map<String, Any?> {
+        return when (locationError) {
+            NotRunning -> {
+                mapOf(KEY_TYPE to TYPE_LOCATION_ERROR_NOT_RUNNING)
+            }
+            Starting -> {
+                mapOf(KEY_TYPE to TYPE_LOCATION_ERROR_STARTING)
+            }
+            is Errors -> {
+                mapOf(
+                    KEY_TYPE to TYPE_LOCATION_ERROR_ERRORS,
+                    KEY_VALUE to locationError.errors.map { serializeHypertrackError(it) }
+                )
+            }
+        }
+    }
+
+    fun serializeHypertrackError(error: HyperTrackError): Map<String, String> {
+        return mapOf(
+            KEY_TYPE to TYPE_HYPERTRACK_ERROR,
+            KEY_VALUE to error.name
+        )
+    }
+
+    fun serializeDeviceId(deviceId: String): Map<String, Any?> {
+        return mapOf(
+            KEY_TYPE to TYPE_DEVICE_ID,
+            KEY_VALUE to deviceId
+        )
+    }
+
+    fun deserializeDeviceName(name: Map<String, Any?>): Result<String> {
+        return parse(name) {
+            it.assertValue<String>(key = KEY_TYPE, value = TYPE_DEVICE_NAME)
+            it.get<String>(KEY_VALUE).getOrThrow()
+        }
+    }
+
+    fun deserializeAvailability(isAvailable: Map<String, Any?>): Result<Boolean> {
+        return parse(isAvailable) {
+            it.assertValue<String>(key = KEY_TYPE, value = TYPE_AVAILABILITY)
+            it.get<Boolean>(KEY_VALUE).getOrThrow()
+        }
+    }
+
+    fun deserializeGeotagData(map: Map<String, Any?>): Result<Geotag> {
+        return parse(map) {
+            val data = it.get<Map<String, Any?>>(KEY_GEOTAG_DATA).getOrThrow()
+            Geotag(data)
+        }
+    }
+
+    fun <T> parse(
+        source: Map<String, Any?>,
+        parseFunction: (ParsingHandle) -> T
+    ): Result<T> {
+        val handle = ParsingHandle(source);
+        return try {
+            if (handle.exceptions.isEmpty()) {
+                Success(parseFunction.invoke(handle))
+            } else {
+                Failure(ParsingExceptions(source, handle.exceptions))
+            }
+        } catch (e: Exception) {
+            Failure(
+                if (handle.exceptions.isNotEmpty()) {
+                    ParsingExceptions(source, handle.exceptions + e)
+                } else {
+                    e
```

Comment:
this e is not mapped to `ParsingExceptions`

Reply:
Why it should be?